modules/apps/foundation/roles/roles-admin-web/src/main/resources/META-INF/resources/edit_role_permissions_navigation.jspf

```diff
@@ -145,7 +145,9 @@ PanelCategoryRegistry panelCategoryRegistry = (PanelCategoryRegistry)request.get
 											for (PanelApp panelApp : panelApps) {
 												Portlet panelAppPortlet = PortletLocalServiceUtil.getPortletById(company.getCompanyId(), panelApp.getPortletId());
 
-												if (com.liferay.portal.kernel.portlet.OmniadminControlPanelEntry.class.isAssignableFrom(panelAppPortlet.getControlPanelEntryInstance().getClass())) {
+												if (Objects.equals(panelAppPortlet.getControlPanelEntryClass(), com.liferay.portal.kernel.portlet.OmniadminControlPanelEntry.class.getName()) ||
```

Comment:
I wonder if it might be better to show the portlet but disabled and maybe with an explanation help message.

Reply:
That's a good idea, though I think that change would introduce new behavior (rather than fixing a regression as this ticket does).  If we go that route it should be as part of a story ticket, since it would require quite a few changes rather than just fixing this if check.  Do you agree?

Reply:
Good point.

Let's not do it for now and just keep in mind as we work more on this portlet in the future.